crates/oxc_linter/src/rules/jsx_a11y/aria_props.rs

```diff
@@ -0,0 +1,156 @@
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeName},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use std::collections::HashSet;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jsx-a11y(aria-props):")]
+#[diagnostic(severity(warning), help("{1}"))]
+struct AriaPropsDiagnostic(#[label] pub Span, String);
+
+#[derive(Debug, Default, Clone)]
+pub struct AriaProps;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that elements do not use invalid ARIA attributes.
+    ///
+    /// ### Why is this bad?
+    /// Using invalid ARIA attributes can mislead screen readers and other assistive technologies.
+    /// It may cause the accessibility features of the website to fail, making it difficult
+    /// for users with disabilities to use the site effectively.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <input aria-labeledby="address_label" />
+    ///
+    /// // Good
+    /// <input aria-labelledby="address_label" />
+    /// ```
+    AriaProps,
+    correctness
+);
+
+impl Rule for AriaProps {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let valid_aria_props = create_valid_aria_props();
+
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            for attr in &jsx_el.attributes {
+                if let JSXAttributeItem::Attribute(attr) = attr {
+                    let name = get_attribute_name(&attr.name).to_lowercase();
+                    if name.starts_with("aria-") && !valid_aria_props.contains(&name) {
+                        let error_message =
+                            format!("{}: This attribute is an invalid ARIA attribute.", name);
+                        ctx.diagnostic(AriaPropsDiagnostic(attr.span, error_message));
+                    }
+                }
+            }
+        }
+    }
+}
+
+fn get_attribute_name(attr: &JSXAttributeName) -> String {
+    match attr {
+        JSXAttributeName::Identifier(ident) => ident.name.to_string(),
+        JSXAttributeName::NamespacedName(namespaced_name) => {
+            format!("{}:{}", namespaced_name.namespace.name, namespaced_name.property.name)
+        }
+    }
+}
+
+/// Creates a set of valid ARIA properties from the WAI-ARIA 1.1 specifications.
+///
+/// The properties list is based on W3C's WAI-ARIA standards.
+/// Reference: <https://www.w3.org/TR/wai-aria/#state_prop_def>
+///
+/// # Returns
+/// `HashSet<String>` containing valid ARIA properties.
+fn create_valid_aria_props() -> HashSet<String> {
```

Comment:
can we use a [phf set](https://docs.rs/phf/latest/phf/macro.phf_set.html) instead?

it should have better perf

https://github.com/oxc-project/oxc/blob/89b6fce95819c9600c45969429d160be4862d9e6/crates/oxc_linter/src/globals.rs#L85-L90

Reply:
Thanks for pointing that out. I wasn't aware of that. I've also moved this set to globals.rs as well. https://github.com/oxc-project/oxc/pull/1797/commits/3de2f5b1dbf8d7e40617355d267ef0c5dbafb1ce

Reply:
no worries - thanks!